pkg/controllers/machinesync/machine_sync_controller_test.go

```diff
@@ -91,35 +140,236 @@ var _ = Describe("MachineSync Reconciler", func() {
+					mapiMachine.Status.AuthoritativeAPI = ""
+				})).Should(Succeed())
+			})
+
+			It("should not make any changes", func() {
+				resourceVersion := mapiMachine.GetResourceVersion()
+				Consistently(k.Object(mapiMachine), timeout).Should(
+					HaveField("ResourceVersion", Equal(resourceVersion)),
+				)
+			})
+		})
+
+		Context("when the MAPI machine does not exist and the CAPI machine does", func() {
+			BeforeEach(func() {
+				By("Creating the CAPI machine")
+				capiMachine = capiMachineBuilder.Build()
+
+				Expect(k8sClient.Create(ctx, capiMachine)).Should(Succeed())
+			})
+
+			It("should not make any changes to the CAPI machine", func() {
+				resourceVersion := capiMachine.GetResourceVersion()
+				Consistently(k.Object(capiMachine), timeout).Should(
+					HaveField("ResourceVersion", Equal(resourceVersion)),
+				)
+			})
+
+			It("should not create a MAPI machine", func() {
+				Consistently(k.ObjectList(&machinev1beta1.MachineList{}), timeout).ShouldNot(HaveField("Items",
+					ContainElement(HaveField("ObjectMeta.Name", Equal(capiMachine.GetName()))),
+				))
+			})
+		})
+	})
+
+	Context("when the CAPI infra machine resource does not exist", func() {
+		Context("when the MAPI machine has MachineAuthority set to Machine API", func() {
+			BeforeEach(func() {
+				By("Creating MAPI machine")
+				mapiMachine = mapiMachineBuilder.Build()
+
+				Expect(k8sClient.Create(ctx, mapiMachine)).Should(Succeed())
+
+				By("Setting the AuthoritativeAPI to MachineAPI")
+				Eventually(k.UpdateStatus(mapiMachine, func() {
+					mapiMachine.Status.AuthoritativeAPI = machinev1beta1.MachineAuthorityMachineAPI
+				})).Should(Succeed(), "should have succeeded updating the AuthoritativeAPI")
+			})
+
+			Context("when the CAPI machine does not exist", func() {
+				It("should create the CAPI machine", func() {
+					capiMachine = capiv1resourcebuilder.Machine().WithName(mapiMachine.Name).WithNamespace(capiNamespace.Name).Build()
+					Eventually(k.Get(capiMachine)).Should(Succeed(), "should have succeeded getting a CAPI Machine")
+				})
+
+				It("should create the CAPI infra machine", func() {
+					capiInfraMachine := capav1builder.AWSMachine().WithName(mapiMachine.Name).WithNamespace(capiNamespace.Name).Build()
+					Eventually(k.Get(capiInfraMachine)).Should(Succeed(), "should have succeeded getting a CAPI Infra Machine")
+				})
+
+				It("should update the synchronized condition on the MAPI machine to True", func() {
+					Eventually(k.Object(mapiMachine), timeout).Should(
+						HaveField("Status.Conditions", ContainElement(
+							SatisfyAll(
+								HaveField("Type", Equal(consts.SynchronizedCondition)),
+								HaveField("Status", Equal(corev1.ConditionTrue)),
+								HaveField("Reason", Equal("ResourceSynchronized")),
+								HaveField("Message", Equal("Successfully synchronized MAPI Machine to CAPI")),
+							))),
+					)
+				})
+			})
+
+			Context("when the CAPI machine does exist", func() {
```

Comment:
What about when the infra machine exists but not the normal machine?

Reply:
We do have a case for this right? 🤔 
Line 197 of the that file.

Reply:
Ack yep, I see it now